src/spanningtrees/kruskal.jl

```diff
@@ -10,7 +10,7 @@ function kruskal_mst end
 
     connected_vs = IntDisjointSets(nv(g))
 
-    mst = Vector{Edge}()
+    mst = Vector{AbstractEdge}()
```

Comment:
On second thought, I think this makes the function type-unstable. Can you show the output of `@code_warntype`?

Reply:
Here it is:
```julia
@code_warntype kruskal_mst(SimpleWeightedGraph([0 2; 2 0]))
```
> Body::Array{AbstractEdge,1}
> │╻╷ weights350 1 ─ %1 = (Base.getfield)(g, :weights)::SparseArrays.SparseMatrixCSC{Int64,Int64}
> │╻  kruskal_mst    │   %2 = invoke LightGraphs.kruskal_mst(SimpleTraits.Not{IsDirected{SimpleWeightedGraph{Int64,Int64}}}::Type{SimpleTraits.Not{IsDirected{SimpleWeightedGraph{Int64,Int64}}}}, _2::SimpleWeightedGraph{Int64,Int64}, %1::SparseArrays.SparseMatrixCSC{Int64,Int64})::Array{AbstractEdge,1}
> │      └──      return %2

Reply:
So I don't think we should be returning a vector of an abstract type. What if we did
```
mst = Vector{edgetype(g}}()
```
instead?

Reply:
It works well, now the type is concrete:
> Body::Array{SimpleWeightedEdge{Int64,Int64},1}
>    weights350 1 ─ %1 = (Base.getfield)(g, :weights)::SparseArrays.SparseMatrixCSC{Int64,Int64}
>     kruskal_mst    │   %2 = invoke LightGraphs.kruskal_mst(SimpleTraits.Not{IsDirected{SimpleWeightedGraph{Int64,Int64}}}::Type{SimpleTraits.Not{IsDirected{SimpleWeightedGraph{Int64,Int64}}}}, _2::SimpleWeightedGraph{Int64,Int64}, %1::SparseArrays.SparseMatrixCSC{Int64,Int64})::Array{SimpleWeightedEdge{Int64,Int64},1}
>                 return %2

Pushed fix.